src/common/curl_tools.c

```diff
@@ -28,6 +28,10 @@ void dt_curl_init(CURL *curl, gboolean verbose)
 {
   curl_easy_reset(curl);
 
+  char useragent[64];
```

Comment:
Really there is no need for magic constants or dynamic allocations:
```
const char prefix[] = “darktable/“;
const char useragent[sizeof(prefix) + sizeof(darktable_version_string) - 1];
snprintf(useragent, sizeof(useragent), “%s%s”, prefix, darktable_version_string);
```

Reply:
@parafin : Even better indeed, I missed the fact that the version string was static. So yes the solution to use. Thanks!

Reply:
`sizeof(darktable_package_version)` isn't available in this context at compile time. The variable is declared as an `extern char[]`  in `config.h` while the actual string lives in `version_gen.c`.

An ideal workaround would be generating version string literals directly in `config.h` header in the first place, and by extension the entire UA string can reside in rodata.

Reply:
I missed the fact that it was extern. Having version string in a separate object file have a benefit of faster project recompilation after a version change. I would suggest to calculate the string length at runtime then, I think C even allows array size on stack to be dynamic.